v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SpannerTransactionWriter.java

```diff
@@ -160,6 +177,8 @@ private static Result create(
 
     public abstract PCollection<FailsafeElement<String, String>> retryableErrors();
 
+    public abstract PCollection<String> filteredEvents();
+
```

Comment:
I thought about this a bit more and IMO this shouldn't be here at all?

The `Result` captured here captures what happens as a result of an attempt to write Mutations to Spanner, ie. success, retryable and permanent errors.

Filtered events is something that has already happened **before** a JsonNode is even converted to a mutation, right? Looks like we are overloading this logic.

Reply:
Done. Added transformation as a separate step